trillian/ctfe/cache/cache.go

```diff
@@ -0,0 +1,21 @@
+package cache
+
+import "context"
+
+// NonLeafCertificateChainCache is an interface which allows CTFE binaries to use different cache implementations for non-leaf certificate chains.
+type NonLeafCertificateChainCache interface {
+	// IsLazyLoading returns whether lazy loading is enabled.
+	IsLazyLoading() bool
+
+	// Contains returns whether the hash exists in the cache.
+	Contains(ctx context.Context, hash string) (bool, error)
+
+	// Get returns the non-leaf certificate chain associated with the provided hash.
+	Get(ctx context.Context, hash string) ([]byte, error)
+
+	// Set inserts the key-value pair of non-leaf certificate chain.
+	Set(ctx context.Context, hash string, chain []byte) error
+
+	// Purge clears all data in the cache.
+	Purge(ctx context.Context) error
```

Comment:
Is there a use-case for this? If needed, I would consider having this as a method on the implementation but not necessarily the interface. The same could be true of the lazyloading method. In general, keep interfaces as tight as possible. You could always add an extended interface if needed (e.g. `CacheAdmin` for these rarer operations).

Reply:
I've removed the `Purge` method since there is no use-case at the moment.